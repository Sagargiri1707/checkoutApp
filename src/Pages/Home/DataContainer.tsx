import React, { lazy, useContext, useCallback, Suspense } from 'react';
import { purchaseItem } from '../../reducers/Actions';
import { GlobalAppContext } from '../../Context/Context';
import { LoaderSVG } from '../../Constants';

const Button = lazy(() => import('../../Components/Button'));
const ProductList = lazy(() => import('../../Components/ProductList'));
const Address = lazy(() => import('../../Components/AddressList'));
const FinalConfirmation = lazy(
  () => import('../../Components/FinalConfirmation')
);
import { changeCurrentStep } from '../../reducers/Actions';
import { withErrorBoundary } from '../../HOC/errorBoundaryHoc';

type currentStepProps = {
  currentStep: number;
};
function DataContainer(props: currentStepProps) {
  const {
    dispatch,
    checkedState,
    currentAddress,
    productDetails,
    successLoader,
  } = useContext(GlobalAppContext);
  const { currentStep } = props;
  const continueToNextStep = useCallback(() => {
    dispatch(changeCurrentStep(currentStep + 1));
  }, [changeCurrentStep, currentStep]);

  let text = 'Continue';
  let isBtnDisabled = false;
  switch (currentStep) {
    case 0:
      isBtnDisabled = checkedState.filter(Boolean).length <= 0;
      break;
    case 1:
      isBtnDisabled = currentAddress !== -1;
      break;
    case 2:
      text = 'Place order';
      break;
  }
  return (
    <React.Fragment>
      {currentStep === 0 && (
        <React.Fragment>
          <Suspense fallback={<LoaderSVG />}>
            <ProductList />
          </Suspense>

          {productDetails?.productData?.length !== 0 && (
            <Suspense fallback={<LoaderSVG />}>
              <Button
                text={text}
                disabled={isBtnDisabled}
                onChange={continueToNextStep}
                id={0}
                color={'rose'}
                customClass={'mt-6 lg:mt-0'}
              />
            </Suspense>
          )}
        </React.Fragment>
      )}
      {currentStep === 1 && (
        <React.Fragment>
          <Suspense fallback={<LoaderSVG />}>
            <Address />
          </Suspense>
          <Suspense fallback={<LoaderSVG />}>
            <Button
              text={text}
              disabled={!isBtnDisabled}
              onChange={continueToNextStep}
              id={0}
              color={'rose'}
            />
          </Suspense>
        </React.Fragment>
      )}
      {currentStep === 2 && (
        <React.Fragment>
          <Suspense fallback={<LoaderSVG />}>
            <FinalConfirmation />
          </Suspense>
          <Suspense fallback={<LoaderSVG />}>
            <Button
              text={successLoader ? 'Loading' : text}
              disabled={false}
              onChange={() => purchaseItem(dispatch)}
              id={0}
              type="button"
              color={'rose'}
              customClass={''}
            />
          </Suspense>
        </React.Fragment>
      )}
    </React.Fragment>
  );
}

export default withErrorBoundary('DataContainer',DataContainer);
