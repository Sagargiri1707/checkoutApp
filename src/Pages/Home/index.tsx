import ProductList from "../../Components/ProductList";

function Home() {
  return <div className="max-w-44">
    <div className="md:container md:mx-auto p-4 m-16 mt-12 border flex-col items-center justify-center flex">
      <ProductList />
    
    </div>
  </div>;
}

export default Home;
