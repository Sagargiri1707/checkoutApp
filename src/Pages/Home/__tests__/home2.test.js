
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Home from '../index'
jest.mock('react', () => {
  const ActualReact = jest.requireActual('react');
  return {
    ...ActualReact,
    useContext: () => ({
        currentStep:1,
        dispatch:jest.fn(),
        checkedState:[true,false],
        currentAddress:0,
        productDetails:{productList:[{
          "id":151,
                "media": {
                  "type":"image",
                  "url": "http://rukmini1.flixcart.com/image/224/224/xif0q/stuffed-toy/r/v/u/extra-large-unicorn-stuffed-toy-animal-unicorn-soft-toys-for-original-imagv4bgwc5jhphr.jpeg?q=90"
                        
                },
                "quantity": 1,
                "ratingValue": {
                    "type": "RatingValue",
                    "average": 4.1,
                    "base": 5,
                    "count": 506,
                    "roundOffCount": "506"
                },
                "titles": {
                    "subtitle": "Purple & Pink",
                    "title": "Macros Extra Large Unicorn Stuffed toy Animal Unicorn Soft Toys for Girls, Teddy Bear  - 100 cm"
                },
                "deliveryDetails":{
                    "eta":"29 Jan 2024",
                    "deliveryCharge":"Rs 40",
                    "isdeliveryChargeWaived":true,
                    "isFasterDeliveryAvailable": true,
                    "time": 6300
                },
                "sellerDetails":{
                    "name":"Muskaan seller"
                },
                "itemPrice":{
                    "originalPrice":999.00,
                    "discountedPrice":199.00,
                    "percentageOff": 80,
                    "offersAvailable":{
                        "count":3
                    }
                }
            }]},
        successLoader:false,
    }),
  };
});
test('renders Home component without crashing', async () => {
  const container = render(
      <Home />
  );
  expect(container).toMatchSnapshot()
});

