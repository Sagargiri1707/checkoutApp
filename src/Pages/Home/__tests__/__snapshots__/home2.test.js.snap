// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-44 lg:max-w-96 flex items-center justify-center"
      >
        <div
          class="relative lg:container md:mx-auto lg:p-4 m-0 lg:m-4  mt-2 md:mt-12   flex-col items-center justify-center flex"
        >
          <button
            class="absolute left-0 top-0 bg-blue-300 hover:bg-blue-400 text-white font-bold py-2 px-4 rounded transform scale-x-[-1]"
          >
            <svg
              aria-hidden="true"
              class="w-5 h-5"
              fill="none"
              viewBox="0 0 14 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 5h12m0 0L9 1m4 4L9 9"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] text-primary motion-reduce:animate-[spin_1.5s_linear_infinite]"
            role="status"
          >
            <span
              class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
            >
              Loading...
            </span>
          </div>
          <div
            class="inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] text-primary motion-reduce:animate-[spin_1.5s_linear_infinite]"
            role="status"
          >
            <span
              class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-44 lg:max-w-96 flex items-center justify-center"
    >
      <div
        class="relative lg:container md:mx-auto lg:p-4 m-0 lg:m-4  mt-2 md:mt-12   flex-col items-center justify-center flex"
      >
        <button
          class="absolute left-0 top-0 bg-blue-300 hover:bg-blue-400 text-white font-bold py-2 px-4 rounded transform scale-x-[-1]"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            fill="none"
            viewBox="0 0 14 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 5h12m0 0L9 1m4 4L9 9"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] text-primary motion-reduce:animate-[spin_1.5s_linear_infinite]"
          role="status"
        >
          <span
            class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
          >
            Loading...
          </span>
        </div>
        <div
          class="inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] text-primary motion-reduce:animate-[spin_1.5s_linear_infinite]"
          role="status"
        >
          <span
            class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
